src/bmp.imageio/bmpinput.cpp

```diff
@@ -299,7 +299,13 @@ BmpInput::read_rle_image()
     int rletype = m_dib_header.compression == RLE4_COMPRESSION ? 4 : 8;
     m_spec.attribute("compression", rletype == 4 ? "rle4" : "rle8");
     m_uncompressed.clear();
-    m_uncompressed.resize(m_spec.height * m_spec.width);
+
+    long size = (long)m_spec.height * m_spec.width;
```

Comment:
Both `int` and `long` are 4-bytes on Windows, so this won't change anything there.  Additionally, there's an existing check on line 195 which implies that the real max is `uint32_t` with an appropriate error message/early-return already.

I believe the fix should be to just do the following:
`m_uncompressed.resize(m_spec.image_pixels());`

Reply:
Thank you.  Your advice was correct.  I have resubmitted it